packages/bitcore-lib/lib/transaction/input/taproot.js

```diff
@@ -40,14 +40,18 @@ TaprootInput.prototype.getSignatures = function(transaction, privateKey, index,
   if (!signature) {
     return [];
   }
-  return [new TransactionSignature({
+  const txSig = new TransactionSignature({
     publicKey: privateKey.publicKey,
     prevTxId: this.prevTxId,
     outputIndex: this.outputIndex,
     inputIndex,
     signature: Signature.fromSchnorr(signature),
     sigtype: sigtype
-  })];
+  });
+  if (!this.isValidSignature(transaction, txSig)) {
```

Comment:
Just a stylistic consideration:

return this.isValidSignature(transaction, txSig) ? [txSig] : []

Makes it clear at a glance that this is the end of the method and the return is based on a conditional.